soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4873,6 +4910,64 @@ CustomMessage Randomizer::GetWarpSongMessage(u16 textId, bool mysterious) {
     return messageEntry;
 }
 
+CustomMessage Randomizer::GetMiscMessage(s16 scene, u16 originalTextId) {
+    CustomMessage messageEntry; 
+    if (originalTextId == TEXT_FROGS_UNDERWATER) {
+        messageEntry = CustomMessageManager::Instance->RetrieveMessage(Randomizer::randoMiscHintsTableID, originalTextId);
+        RandomizerGet frogsGet = this->itemLocations[RC_ZR_FROGS_OCARINA_GAME].rgID;
+        std::array<std::string, LANGUAGE_MAX> frogItemName;
+        if (frogsGet == RG_ICE_TRAP) {
+            frogsGet = this->itemLocations[RC_ZR_FROGS_OCARINA_GAME].fakeRgID;
+            frogItemName = {
+                this->itemLocations[RC_ZR_FROGS_OCARINA_GAME].trickName,
+                this->itemLocations[RC_ZR_FROGS_OCARINA_GAME].trickName,
+                this->itemLocations[RC_ZR_FROGS_OCARINA_GAME].trickName
+            };
+        } else {
+            frogItemName = EnumToSpoilerfileGetName[frogsGet];
+        }
+        messageEntry.Replace("{{item}}", std::move(frogItemName[0]), std::move(frogItemName[1]), std::move(frogItemName[2]));
+        return messageEntry;
+    } else if (originalTextId == TEXT_SHEIK_1 || originalTextId == TEXT_SHEIK_2) {
+        messageEntry = CustomMessageManager::Instance->RetrieveMessage(Randomizer::hintMessageTableID, originalTextId);
+        switch (scene) {
+            case SCENE_TOKINOMA:
+                if (originalTextId == TEXT_SHEIK_1) {
+                    messageEntry.Replace("{{message}}", 
+                    "@,&meet me at %gGanon's Castle%w&once you obtain the %rkey to his lair%w.",
+                    "Test",
```

Comment:
Are these Test strings supposed to be here?

Reply:
They're placeholders for French and German translations. Normally I would use AI-generated (read: DeepL) translations but I just forgot 😂